exporter/elasticsearchexporter/model.go

```diff
@@ -58,9 +73,47 @@ func (m *encodeModel) encodeLog(resource pcommon.Resource, record plog.LogRecord
 		document.Sort()
 	}
 
-	var buf bytes.Buffer
-	err := document.Serialize(&buf, m.dedot)
-	return buf.Bytes(), err
+	return document
+
+}
+
+func (m *encodeModel) encodeLogECSMode(resource pcommon.Resource, record plog.LogRecord, scope pcommon.InstrumentationScope, now time.Time) objmodel.Document {
+	var document objmodel.Document
+
+	// First, try to map resource-level attributes to ECS fields.
+	resourceAttrsConversionMap := map[string]string{
+		semconv.AttributeServiceInstanceID:     "service.node.name",
+		semconv.AttributeDeploymentEnvironment: "service.environment",
+		semconv.AttributeTelemetrySDKName:      "agent.name",
+		semconv.AttributeTelemetrySDKVersion:   "agent.version",
+		semconv.AttributeTelemetrySDKLanguage:  "service.language.name",
+		semconv.AttributeCloudPlatform:         "cloud.service.name",
+		semconv.AttributeContainerImageTags:    "container.image.tag",
+		semconv.AttributeHostName:              "host.hostname",
+		semconv.AttributeHostArch:              "host.architecture",
+		semconv.AttributeProcessExecutablePath: "process.executable",
+		semconv.AttributeOSType:                "os.platform",
```

Comment:
Should we be vetting the values here? ECS expects `os.platform` to have one of a restricted set of values -- https://www.elastic.co/guide/en/ecs/current/ecs-os.html#field-os-type

For iOS and Android, I think we should set `os.platform` from `os.name` ("Android" or "iOS") rather than `os.type` ("linux" or "darwin")

See https://github.com/elastic/apm-data/blob/0e072ceaadae2ef8f61d75c3a27c47cc5ce81257/input/otlp/metadata.go#L320-L338

Reply:
Implemented in d36a76fff8.